### ISSUE TITLE
Scrolling back up on Android can sporadically freeze the app 

### ISSUE THREAD

**Issue by mvtglobally** [Expensify/App#11159]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Open any long chat conversation 
2. Scroll back up  
3. Go back and forth from the chat and LHN 

## Expected Result:
app should not freeze 

## Actual Result:
app freezes and doesn't load content for some time 
Just go in and out of a chat repeatedly, back and forth into the same chat, and then back to the LHN. On Android native it will gradually slow down to a crawl, even though clearly there's no network happening, and nothing should be happening except for pure animation. There's some kind of fundamental problem under the hood there.

## Workaround:
unknown 

## Platform:

Where is this issue occurring?

- Android

**Version Number:** 1.2.3-0
**Reproducible in staging?:** Y 
**Reproducible in production?:** Y 
**Email or phone of affected tester (no customers):** 
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation 
 
https://user-images.githubusercontent.com/43995119/191411714-d756af09-3743-4e06-850a-5b586a7a0f0b.mp4

https://user-images.githubusercontent.com/43995119/191411696-70558350-8f62-44a2-a7a6-3616f140bc69.mp4
 
**Expensify/Expensify Issue URL:** 
**Issue reported by:** @JmillsExpensify @quinthar 
**Slack conversation:** https://expensify.slack.com/archives/C01GTK53T8Q/p1663187471835049  
https://expensify.slack.com/archives/C01GTK53T8Q/p1663339978134359

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by gagtic** [comment]
are you using a Flatlist to render? if so, make sure you are properly following the rules of the optimized flatlist. Also make sure that the row of flatlist does not have their own states, should be managed by the parent screen.

I also use React.memo for rendering rows and it does help with the performance, but it sometimes also causes unexpected behavior, so use it with precaution and on your own risk:
Example: export default React.memo(ReceieverRow)

Link for flatlist optimization by react native team:
https://reactnative.dev/docs/optimizing-flatlist-configuration

**Comment by Christinadobrzyn** [comment]
Looks like this was reported here - https://github.com/Expensify/App/issues/8150 - sounds like we decided to 'do nothing' on the other issue so leaving this to eng to decide if we should do something now.

**Comment by jasperhuangg** [comment]
To clarify, are we referring to how scrolling upwards stops at times while new content is being rendered above the position we've scrolled to? If so, then I'm not sure there's much we can do here, other chat apps also experience the same types of issues. Even though there aren't any network requests, I don't think the content has been rendered yet, which means the scrolling has to stop while the content is rendered before we can continue.

I'm going to drop a note in #engineering-chat to get a second opinion, perhaps I'm understanding the issue incorrectly?

**Comment by jasperhuangg** [comment]
Discussion happening here: https://expensify.slack.com/archives/C03TQ48KC/p1663789199187759

**Comment by jasperhuangg** [comment]
Closing in favor of https://github.com/Expensify/App/issues/11241
